src/prop/cadical.cpp

```diff
@@ -1190,6 +1227,11 @@ void CadicalSolver::initialize(context::Context* context,
   d_proxy = theoryProxy;
   d_propagator.reset(new CadicalPropagator(theoryProxy, context, *d_solver));
   d_solver->connect_external_propagator(d_propagator.get());
+  if (d_env.getOptions().base.pluginNotifySatClauseInSolve)
```

Comment:
This isn't the right option to check (it controls a detail about when the plugins start listening to clauses).

At the moment, you would always have to register this callback.  Do you think this would be an issue for performance? If so, we should maybe introduce an option e.g. `enablePlugins` which would guard when plugins are allowed.

Reply:
We could also check whether plugins were registered, and only then we enable it?

Reply:
Good point, the plugins are available in `d_env.getPlugins`